### PR TITLE
Replace all plain occurrences of $gray-dark with CSS variables

### PR DESCRIPTION
**Pull request by frontdevde**
#### Changes proposed in this Pull Request

* Replace all plain occurrences of $gray-dark with CSS variables
* ~~Disable stylelint selector-max-id rule on _main.scss~~

#### Testing instructions

* Navigate the calypso.live link and check if everything matches the current staging/production environments

### CODE REVIEW

**Comment by drw158** [comment]
Not a blocker, but if the css property is `color` (assuming the color is being applied to text), then it should really be using the `--color-text` variable, which is equivalent to the 800 neutral value. 

Not a huge deal since these gray-dark variables weren't using the color text scss variable before.

**Comment by drw158** [comment]
The more I think about it, I think it would be great to use --color-text. That way, we can consolidate text colors (700 and 800) into one value. We could do that after merging this though.

File: client/my-sites/media-library/list-item.scss [hunk 1]
```diff
@@ -71,19 +74,22 @@
 .media-library__list-item.is-transient .media-library__list-item-figure::after {
 	content: '';
 	position: absolute;
-		top: 0;
-		right: 0;
-		bottom: 0;
-		left: 0;
+	top: 0;
+	right: 0;
```
Comment (torres126) [inline client/my-sites/media-library/list-item.scss]: Just to note, the style guidelines do advocate the indentation here under the "Positioning" section. I wonder if that should be changed though, it's rare to see it for some reason.

Comment (flootr) [reply client/my-sites/media-library/list-item.scss]: Yep, I think that's introduced by running prettier locally on those files. Let's see if stylelint can fix those.

**Pull request merged**